### PR TITLE
S112.14 add time helper

### PR DESCRIPTION
**Pull request by kulltc**

### CODE REVIEW

**Comment by expand-dev** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [TimeUtil.php](https://github.com/ExpandOnline/CakePHPUtil/blob/2c544233d02c33b6135c8f211a44bd0595a9f3c9/Lib/Time/TimeUtil.php): Replace all tab characters in this file by sequences of white-spaces. [rule](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS105)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Lib/Time/TimeUtil.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class TimeUtil {
+
+	public function hoursToDecimal($hours) {
+		$hours = trim($hours);
+		if (is_null($hours)) {
+			return '';
+		}
+		$sign = 1;
+		if (substr($hours,0, 1) === '-') {
```
Comment (pvaass) [inline Lib/Time/TimeUtil.php]: Spatie na de komma

File: Test/Case/Lib/TimeUtilTest.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+use CakePHPUtil\Lib\Time\TimeUtil;
+
+class TimeUtilTest extends CakeTestCase {
+
+	/**
+	 * @var TimeUtil
+	 */
+	public $TimeUtil;
```
Comment (pvaass) [inline Test/Case/Lib/TimeUtilTest.php]: Dit is geen model toch? Kleine letter T dan

File: Lib/Time/TimeUtil.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace CakePHPUtil\Lib\Time;
+
+/**
+ * Class TimeHelper
+ * Put time manipulation methods here.
+ */
+
```
Comment (pvaass) [inline Lib/Time/TimeUtil.php]: Newline hier weg

File: Lib/Time/TimeUtil.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class TimeUtil {
+
+	public function hoursToDecimal($hours) {
```
Comment (pvaass) [inline Lib/Time/TimeUtil.php]: Ik dacht dat jullie docblocks wilden :DDD

**Comment by kulltc** [comment]
@pvaass done

File: Lib/Time/TimeUtil.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		$parts = explode(':', $hours);
+		foreach ([
+			'hours',
+			'minutes',
+			'seconds'
+		] as $index => $name) {
+			${$name} = $parts[$index] ?? 0;
+		}
+
+		return $sign * ($hours + ($minutes / 60) + (($seconds ?? 0) / 3600));
```
Comment (expand-dev) [inline Lib/Time/TimeUtil.php]: MINOR Remove this unused "$minutes" local variable. [rule](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS1481)
MINOR Remove this unused "$seconds" local variable. [rule](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS1481)

**Comment by expand-dev** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [TimeUtil.php](https://github.com/ExpandOnline/CakePHPUtil/blob/5467d54613a57c05bd63a6740be80451a686fc58/Lib/Time/TimeUtil.php): Replace all tab characters in this file by sequences of white-spaces. [rule](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS105)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by expand-dev** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [TimeUtil.php](https://github.com/ExpandOnline/CakePHPUtil/blob/a0befcd1ff270ec8a61b95c4691f23981b39ccaa/Lib/Time/TimeUtil.php): Replace all tab characters in this file by sequences of white-spaces. [rule](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS105)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**